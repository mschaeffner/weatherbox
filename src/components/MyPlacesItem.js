import React from 'react';

export default ({place, onSelect}) =>
  <a key={place.id} style={styles.link} onClick={() => onSelect(place.id)}>
    {place.name}
  </a>

const styles = {
  link: {
    cursor: 'pointer',
    display: 'block'
  }
}
