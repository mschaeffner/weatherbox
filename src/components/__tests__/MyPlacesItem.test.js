import React from 'react';
import renderer from 'react-test-renderer';
import MyPlacesItem from '../MyPlacesItem';


describe('MyPlacesItem', () => {
  it('should show the name of the place as a link', () => {

    const place = {
      name: 'Berlin',
      id: 123
    }

    const component = renderer.create(<MyPlacesItem place={place} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
