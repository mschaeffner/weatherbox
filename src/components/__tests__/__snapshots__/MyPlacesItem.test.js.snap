// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyPlacesItem should show the name of the place as a link 1`] = `
<a
  onClick={[Function]}
  style={
    Object {
      "cursor": "pointer",
      "display": "block",
    }
  }
>
  Berlin
</a>
`;
